app/forms/candidate_interface/degree_wizard.rb

```diff
@@ -461,6 +420,75 @@ def specified_grades?
 
   private
 
+    def reviewing_or_country_not_changed_steps(step)
+      case step
+      when :degree_level
+        degree_has_type? ? :type : :review
+      when :completed
+        :award_year
+      when :award_year
+        completed? && international? ? :enic : :review
+      when :enic
+        if international?
+          enic_reason == HAS_STATEMENT ? :enic_reference : :review
+        else
+          :review
+        end
+      else
+        :review
+      end
+    end
+
+    def uk_steps(step)
+      return :type if step == :subject && degree_has_type?
+      return :start_year if step == :completed && phd?
+
+      next_step = {
+        country: :degree_level,
+        degree_level: :subject,
+        subject: :university,
+        type: :university,
+        university: :completed,
+        completed: :grade,
+        grade: :start_year,
+        start_year: :award_year,
+        award_year: :review,
+      }
+
+      next_step[step] || handle_invalid_step
+    end
+
+    def international_steps(step)
+      return :type if step == :country && country.present?
+      return :start_year if step == :completed && phd?
+      return :enic if step == :award_year && completed?
+      return :enic_reference if step == :enic && enic_reason == HAS_STATEMENT
+
+      next_step = {
+        type: :subject,
+        degree_level: :subject,
+        subject: :university,
+        university: :completed,
+        completed: :grade,
+        grade: :start_year,
+        start_year: :award_year,
+        award_year: :review,
+        enic: :review,
+        enic_reference: :review,
+      }
+
+      next_step[step] || handle_invalid_step
```

Comment:
Same as above, can we use the same pattern for deciding a step each time? It's difficult to move between the different methods.

Reply:
See: https://github.com/DFE-Digital/apply-for-teacher-training/pull/10142#discussion_r1870183336